### DISCUSSION TITLE
My current situation and the search for further maintainers

### DISCUSSION THREAD

**Discussion by doomlaur** [xlnt-community/xlnt#32] [General]
I just wanted to post an update about my own situation at the moment and the reason I've been somewhat inactive during most of October.

First of all, as I already explained in https://github.com/tfussell/xlnt/issues/748, the project I've been working on where I used XLNT is going to end - and that is at the end of December 2024. Due to personal reasons, I also decided to change my job, so I won't work anymore at my employer after November 6. This is, by the way, the reason why I've been very busy over the last couple of weeks. For the future, I would like to concentrate on Rust, so I'm trying to get away from C/C++ due to multiple personal reasons.

In other words, I have good news and bad news. The bad news: I'm soon no longer going to use XLNT in any bigger project. The good news: after November 6, I will have some more spare time to finally finish my work on the last couple of changes I planned for XLNT, which I unfortunately did not have time to finish.

Due to the fact that I'm neither going to use XLNT in a bigger project, nor use XLNT in the future due to focusing on other technologies, I would appreciate if some more people could volunteer to take over the work together with @m7913d and keep XLNT maintained and alive, as it deserves to be.

When it comes to me, I would not like to say that I'm stepping down as a maintainer - at least certainly not until @m7913d finds some other people to help. However, I won't be able to test XLNT as well as I did before (in a more complex environment), so while I will be able to help with my C++ experience, I might not be able to keep up the same quality of work (due to testing on rather simple examples / test projects).

Thank you in advance for your understanding. Dear community, please help us with this important work! Let's work together and make sure that the great XLNT library will continue to exist for a long time!

**Comment by m7913d** [comment]
@doomlaur I wish you all the best for the future of your career and I look forward to the cooperation that remains for us. Thank you for all your help in getting this clone off the ground!

**Reply by doomlaur** [reply]
Thank you for your kind words - it was a pleasure working with you and I'm looking forward to continuing working together :smiley:

**Comment by m7913d** [comment]
@doomlaur Finding other maintainers (or even PR contributors) doesn't seem to be easy (which is not really unexpected). Therefore, I think we should really be open-minded about the future of this project. One option may be to join our efforts with other similar (open-source) projects. The most promising project I'm aware of is [OpenXLSX](https://github.com/troldal/OpenXLSX). Have you ever used/tried this library? Some other libraries were discussed by the OpenXLSX creator [here](https://github.com/troldal/OpenXLSX/blob/c787b61520d2cc3b17134b5a974e5232b2af5fb3/README.md#motivation). Are you aware of other similar libraries?

Based on the github project stars history, OpenXLSX seems to have more attraction recently:

![](https://github.com/user-attachments/assets/ae057639-435e-4c95-bec2-5d0f637ed65e)

Since ~1 year, the OpenXLSX project is basically maintained by a single contributor (the original creator of the library isn't very active). Do you think it is a good idea to contact the OpenXLSX maintainer(s) to discuss the future of both projects?

Before making a decision on the long term future of this project, I think it is necessary to conduct a thorough review of both projects concerning their strengths, design goals, features, performance,... .

What do you think?

_Note that it is not my intention to abandon this project in the (near) feature, but I just want to update you in an early stage on a possible solution for the maintaining problem I'm thinking about. The outcome may be as well that both projects keep to exist along each other._

**Reply by doomlaur** [reply]
> The most promising project I'm aware of is [OpenXLSX](https://github.com/troldal/OpenXLSX). Have you ever used/tried this library?

I know about OpenXLSX. I found this library when I was doing research about the Office Open XML library best suited for the project I was working on. That was about 4 years ago. Back then, I decided to not use OpenXLSX because while it looked like it is in active development and seemed promising, it also seemed to not be production-ready yet (I did not test it, though). In the meantime, OpenXLSX has been further developed, but I don't know more details or anything else about it. I'm sure it's an amazing library!

> Some other libraries were discussed by the OpenXLSX creator [here](https://github.com/troldal/OpenXLSX/blob/c787b61520d2cc3b17134b5a974e5232b2af5fb3/README.md#motivation). Are you aware of other similar libraries?

That list sums up really well the libraries I looked at. I must admit that I did not have the time to test all of them back then, and unfortunately cannot remember any details - sorry! I can just remember that, in the end, XLNT looked like the largest and most developed C++ Excel library at the time, which also provided very programmer-friendly interfaces - which is why I chose to use it in my project (and created multiple PRs now and then).

> Since ~1 year, the OpenXLSX project is basically maintained by a single contributor (the original creator of the library isn't very active). Do you think it is a good idea to contact the OpenXLSX maintainer(s) to discuss the future of both projects?

Sure, feel free to do so! :wink: 

> Before making a decision on the long term future of this project, I think it is necessary to conduct a thorough review of both projects concerning their strengths, design goals, features, performance,... .

Agreed.

> What do you think?
> 
> _Note that it is not my intention to abandon this project in the (near) feature, but I just want to update you in an early stage on a possible solution for the maintaining problem I'm thinking about. The outcome may be as well that both projects keep to exist along each other._

Due to the fact that I'm neither working on a project that needs a library like XLNT (neither now nor in the future), nor even working with C++ anymore, I think it would be unfair from my side to demand anything. I think you're a great programmer and maintainer, and you will be able to make the right decision. I will try to help with XLNT until you find a new maintainer (if possible), as much as I can. Otherwise feel free to go in the direction you prefer the most - you have my full support! :hugs:

**Reply by m7913d** [reply]
> I think it would be unfair from my side to demand anything

But it's nice to have a second opinion. So thanks for your reply!
